src/util/ui.hpp

```diff
@@ -46,5 +46,9 @@ namespace util::ui {
 
     PopupLayout getPopupLayout(cocos2d::CCSize popupSize);
 
-    cocos2d::CCNode* findChildByMenuSelectorRecursive(cocos2d::CCNode* node, uintptr_t function);
+    cocos2d::CCNode* findChildByMenuSelectorRecursive(cocos2d::CCNode* node, uintptr_t function);   
+
+    uibuilder::Build<cocos2d::CCSprite> createBadge(std::string badgePNG, cocos2d::CCPoint pos, std::string id);
+
+    uibuilder::Build<cocos2d::CCSprite> createBadgeIfSpecial(cocos2d::ccColor3B color, cocos2d::CCPoint pos);
```

Comment:
again, imo it would be much cleaner for those methods to just directly return `CCSprite*` instead of `Build<CCSprite>`

also, `pos` argument should be removed and instead managed by the caller. it doesn't make much sense to me for the badge to set its own position; the node that creates the badge should do that itself.

also change `std::string` to `const char*` for better efficiency :p